nebula/ui/components/MZHelpSheet.qml

```diff
@@ -0,0 +1,240 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.15
+import QtQuick.Controls 2.14
+import QtQuick.Layouts 1.14
+
+import components 0.1
+import Mozilla.Shared 1.0
+
+//Usage: pass a list of blocks into the model to instantiate components
+//Fields:
+//"type": MZHelpSheet.BlockType (eg MZHelpSheet.BlockType.PrimaryButton)
+//"text": label or button text (eg "Hello world")
+//"margin": top margin above the component (eg 16)
+//"action": action performed on button click - only applicable when type is MZHelpSheet.BlockType.PrimaryButton or MZHelpSheet.BlockType.LinkButton
+//(eg () => { MZUrlOpener.openUrl("https://mozilla.org") })
+/*
+  MZHelpSheet {
+      title: "MZHelpSheet"
+      iconSource: "qrc:/ui/resources/connection-info-dark.svg"
+
+      model: [
+          {type: MZHelpSheet.BlockType.Title, text: "title"},
+          {type: MZHelpSheet.BlockType.Text, margin: 8, text: "text"},
+          {type: MZHelpSheet.BlockType.Text, margin: 16, text: "text"},
+          {type: MZHelpSheet.BlockType.Text, margin: 16, text: "text"},
+          {type: MZHelpSheet.BlockType.PrimaryButton, text: "Primary", margin: 16, action: () => { sheet.close(); MZNavigator.requestScreen(VPN.ScreenGetHelp) } },
+          {type: MZHelpSheet.BlockType.LinkButton, text: "Link", margin: 8, action: () => { MZUrlOpener.openUrl("https://mozilla.org") } }
+      ]
+    }
+*/
+
+MZBottomSheet {
+    id: bottomSheet
+
+    property alias iconSource: icon.source
+    property alias title: title.text
+    property var model
+
+
+    enum BlockType {
+        Title,
+        Text,
+        PrimaryButton,
+        LinkButton
+    }
+
+    implicitHeight: Math.min(contentItem.implicitHeight, maxHeight)
+
+    contentItem: ColumnLayout {
+        spacing: 0
+
+        ColumnLayout {
+            id: headerLayout
+
+            Layout.topMargin: 8
```

Comment:
Should the constants in this file be moved to theme.js?

Reply:
In my opinion, not every integer literal needs naming. I think if there is no way to genericize an integer for reusability, theres not much of a point in adding it to theme.js

Reply:
But I will go through this component and add constants where I see fit